### DISCUSSION TITLE
Oneshot mappings

### DISCUSSION THREAD

**Discussion by crabdancing** [xremap/xremap#608] [Ideas]
In `keyd`, you can do e.g. `shift = "oneshot(shift)`. This allows you to hit the shift key, and then shift will be locked on until the next keypress, similar to the key behavior you see e.g. in mobile keyboards. Can this be done in xremap?

**Comment by k0kubun** [comment]
It doesn't exist at this moment, but I'm open to accepting that feature.

The behavior is straightforward for modifier keys, but what happens when you use a non-modifier key, e.g. `a = oneshot(a)`? If you press a -> b, would it (1) press A, (2) press B, (3) release B, and (4) release A as well? Does it work while there's no use for it?
